### DISCUSSION TITLE
contact angle influence

### DISCUSSION THREAD

**Discussion by baharkhs** [PMEAL/OpenPNM#2869] [Q&A]
Hi,

I am using drainage algorithm to receive capillary pressure curves in cubic networks. There's one thing I don't understand.
In the context of petroleum engineering I know that the contact angle higher than 90 means the network is oil wet and contact angle lower than 90 means the network is water wet. I wanted to build a network and have two phases of water and oil (air) and I wanted to have the oil as the non-wetting or displacing fluid and the water as the wetting phase, I put also the contact angle to 50 for example. But in drainage algorithm I could not receive the pc_curve. it was like empty. then I tried the invasion percolation and the pc curves were negative.
when I checked the tutorial for this, I saw that you are assigning 140 for the contact angle and using the air in the drainage algorithm. I wanna know why is this threshold of 90 changing the positivity and negativity? and why you are using 140 and using air as the non-wetting phase?

**Comment by jgostick** [comment]
Drainage refers to the invasion of a non-wetting fluid to displace the wetting one.  In the drainage algorithm you specify the properties of the non-wetting fluid.  If you have the wetting fluid displacing the NON-wetting one, then you have imbibition and using a drainage algorithm is not correct.  Imbibition is MUCH hard to get correct so we don't offer an imbibition algorithm.  It would give people the false impression that they just need to do `imb.run()` and get the answers.  Imbibition requires a lot more care and knowledge.
